includes/admin/reports/class-wc-report-downloads.php

```diff
@@ -0,0 +1,251 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+if ( ! class_exists( 'WP_List_Table' ) ) {
+	require_once( ABSPATH . 'wp-admin/includes/class-wp-list-table.php' );
+}
+
+/**
+ * WC_Report_Downloads.
+ *
+ * @author      WooThemes
+ * @category    Admin
+ * @package     WooCommerce/Admin/Reports
+ * @version     3.3.0
+ */
+class WC_Report_Downloads extends WP_List_Table {
+
+	/**
+	 * Max items.
+	 *
+	 * @var int
+	 */
+	protected $max_items;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+
+		parent::__construct( array(
+			'singular'  => 'download',
+			'plural'    => 'downloads',
+			'ajax'      => false,
+		) );
+	}
+
+	/**
+	 * Don't need this.
+	 *
+	 * @param string $position
+	 */
+	public function display_tablenav( $position ) {
+
+		if ( 'top' !== $position ) {
+			parent::display_tablenav( $position );
+		}
+	}
+
+	/**
+	 * Output the report.
+	 */
+	public function output_report() {
+
+		$this->prepare_items();
+
+		// Subtitle for permission if set
+		if ( isset( $_GET['permission_id'] ) && is_numeric( $_GET['permission_id'] ) && $_GET['permission_id'] > 0 ) {
```

Comment:
Rather than repeating the use of `$_GET['permission_id']` throughout this method, maybe we should assign to a variable instead and use that.